aggregators/aggregator_test.go

```diff
@@ -1075,163 +1075,307 @@ func TestHarvest(t *testing.T) {
+				},
+				NumericLabels: modelpb.NumericLabels{
+					"user_id":     &modelpb.NumericLabelValue{Global: true, Value: 100},
+					"cost_center": &modelpb.NumericLabelValue{Global: true, Value: 10},
+				},
+				Metricset: &modelpb.Metricset{
+					Name:     "service_transaction",
+					DocCount: 1,
+					Interval: "1s",
 				},
 			},
-			Transaction: &modelpb.Transaction{
-				Type: "txtype",
-				DurationSummary: &modelpb.SummaryMetric{
-					Count: 1,
-					Sum:   100351, // Estimate from histogram
+		}
+
+		runTest(t, input, expected)
+	})
+
+	t.Run("with multiple global labels values", func(t *testing.T) {
+		input := modelpb.Batch{
+			{
+				Event: &modelpb.Event{
+					Outcome:  "success",
+					Duration: uint64(txnDuration),
+				},
+				Transaction: &modelpb.Transaction{
+					Name:                "foo",
+					Type:                "txtype",
+					RepresentativeCount: 1,
+				},
+				Service: &modelpb.Service{Name: "svc"},
+				Labels: modelpb.Labels{
+					"baz": &modelpb.LabelValue{Global: true, Values: []string{"asd", "qwe"}},
 				},
-				DurationHistogram: &modelpb.Histogram{
-					Values: []float64{100351},
-					Counts: []uint64{1},
+				NumericLabels: modelpb.NumericLabels{
+					"bar": &modelpb.NumericLabelValue{Global: true, Values: []float64{1, 2}},
 				},
 			},
-			Service: &modelpb.Service{
-				Name: "svc",
+			{
+				Event: &modelpb.Event{
+					Outcome:  "success",
+					Duration: uint64(txnDuration),
+				},
+				Transaction: &modelpb.Transaction{
+					Name:                "foo",
+					Type:                "txtype",
+					RepresentativeCount: 1,
+				},
+				Service: &modelpb.Service{Name: "svc"},
+				Labels: modelpb.Labels{
+					"baz": &modelpb.LabelValue{Global: true, Values: []string{"rty", "fgh"}},
+				},
+				NumericLabels: modelpb.NumericLabels{
+					"bar": &modelpb.NumericLabelValue{Global: true, Values: []float64{4, 5}},
+				},
 			},
-			Labels: modelpb.Labels{
-				"department_name": &modelpb.LabelValue{Global: true, Value: "apm"},
-				"organization":    &modelpb.LabelValue{Global: true, Value: "observability"},
-				"company":         &modelpb.LabelValue{Global: true, Value: "elastic"},
+		}
+
+		expected := []*modelpb.APMEvent{
+			{
+				Timestamp: modelpb.FromTime(time.Unix(0, 0).UTC()),
+				Event: &modelpb.Event{
+					SuccessCount: &modelpb.SummaryMetric{
+						Count: 1,
+						Sum:   1,
+					},
+					Outcome: "success",
+				},
+				Transaction: &modelpb.Transaction{
+					Name: "foo",
+					Type: "txtype",
+					Root: true,
+					DurationSummary: &modelpb.SummaryMetric{
+						Count: 1,
+						Sum:   100351, // Estimate from histogram
+					},
+					DurationHistogram: &modelpb.Histogram{
+						Values: []float64{100351},
+						Counts: []uint64{1},
+					},
+				},
+				Service: &modelpb.Service{
+					Name: "svc",
+				},
+				Labels: modelpb.Labels{
+					"baz": &modelpb.LabelValue{Global: true, Values: []string{"asd", "qwe", "rty", "fgh"}},
```

Comment:
label values should not be concatenated

Reply:
Thanks, I updated the code in [`aa210a6` (#191)](https://github.com/elastic/apm-aggregation/pull/191/commits/aa210a60d17e8bc3aa68d6b72c39de1fa91e2d95). That commit still fails because the test was flappy due to difference in sorting on different runs, solved with the next commits.